### PR TITLE
workaround for xarray additional coordinate bug

### PR DESCRIPTION
**Pull request by JGuetschow**
This pr adds a workaround for the issue with additional (non-indexed) coordinates in some xarray functions (https://github.com/pydata/xarray/issues/9124). 

There are also some small improvements of error messages included. 

# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [ ] Documentation added (where applicable) <- not applicable here
- [x] Description in a ``.rst`` file in the directory ``changelog_unreleased`` added - remember to include a ``*`` to make it a bullet point

## Description
The workaround is to add custom `pr.fillna` and `pr.combine_first` methods which wrap around `fillna` and `combine_first` and take care of the additional coordinate. 

The improvements to error messages are to give more relevant information on the data that caused the error.

### CODE REVIEW

File: primap2/_fill_combine.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            except Exception as ex:
+                message = f"Could not re-add lost coordinate {coord}: {ex}"
+                logger.error(message)
+                raise ValueError(message) from ex
+
+        return filled
+
+
+class DatasetFillAccessor(BaseDatasetAccessor):
+    # fillna, combine_first
```
Comment (mikapfl) [inline primap2/_fill_combine.py:104]:
```suggestion
```

File: primap2/_merge.py [hunk 2]
```diff
@@ -88,6 +89,13 @@ def generate_log_message(da_error: xr.DataArray, tolerance: float) -> str:
     * convert the rest into a pandas dataframe for nice printing
     """
     scalar_dims = [dim for dim in da_error.dims if len(da_error[dim]) == 1]
+    scalar_dims_format = []
+    for dim in scalar_dims:
+        if dim == "time":
+            single_date = date.fromtimestamp(da_error[dim].item() / 1000000000)
+            scalar_dims_format.append(f"{dim}={single_date.strftime('%Y')}")
```
Comment (mikapfl) [inline primap2/_merge.py:96]: What's happening here?

**Review by mikapfl — APPROVED** [review]

File: primap2/_fill_combine.py (hunk 1 above)
Comment (JGuetschow) [reply primap2/_fill_combine.py:104]: ups, forgot to delete my todo list

File: primap2/_merge.py (hunk 2 above)
Comment (JGuetschow) [reply primap2/_merge.py:96]: converting the timecode to something human readable (a year). It looks nasty so probably I missed the correct function to use.

Comment (mikapfl) [reply primap2/_merge.py:96]: For now, maybe just add a comment which explains what's going on. I also suspect that it doesn't work if you use another resolution for your datetime, but I don't think it's super important.

Comment (JGuetschow) [reply primap2/_merge.py:96]: But there was also a bug and the formatting was never used. Fixed now

**Pull request merged**